test/integration_test/migration_backup_test.go

```diff
@@ -0,0 +1,177 @@
+	logrus.Infof("Backup path being used: %s", backupLocationPath)
+	t.Run("deploymentMigrationBackupTest", deploymentMigrationBackupTest)
+}
+
+func deploymentMigrationBackupTest(t *testing.T) {
+	for location, secret := range migBackupConfigMap {
+		logrus.Infof("Backing up to cloud: %v using secret %v", location, secret)
+		var err error
+		// Reset config in case of error
+		defer func() {
+			err = setRemoteConfig("")
+			require.NoError(t, err, "Error resetting remote config")
+		}()
+
+		// Trigger migration to backup app from source cluster to destination cluster
+		ctxs, preMigrationCtx := triggerMigration(
+			t,
+			"mysql-migration",
+			"mysql-1-pvc",
+			nil,
+			[]string{"mysql-migration"},
+			true,
+			false,
+			true,
+		)
+
+		// Cleanup up source
+		validateAndDestroyMigration(t, ctxs, preMigrationCtx, true, true, true, false, true)
+
+		logrus.Infoln("Completed migration of apps from source to destination, will now start backup on second cluster")
+
+		// Change kubeconfig to destination cluster
+		err = dumpRemoteKubeConfig(remoteConfig)
+		require.NoErrorf(t, err, "Unable to write clusterconfig: %v", err)
+
+		err = setRemoteConfig(remoteFilePath)
+		require.NoError(t, err, "Error setting remote config")
+
+		// Backup app from the destination cluster to cloud
+		currBackupLocation, err := createBackupLocation(t, appKey+"-backup-location", testKey+"-mysql-migration", storkv1.BackupLocationType(location), secret)
+		require.NoError(t, err, "Error creating backuplocation %s", currBackupLocation.Name)
+
+		currBackup, err := createApplicationBackupWithAnnotation(t, "dest-backup", testKey+"-mysql-migration", currBackupLocation)
+		require.NoError(t, err, "Error creating app backups")
+
+		err = waitForAppBackupCompletion(currBackup.Name, currBackup.Namespace, applicationBackupSyncRetryTimeout)
+		require.NoError(t, err, "Application backup %s in namespace %s failed.", currBackup.Name, currBackup.Namespace)
+
+		destroyAndWait(t, []*scheduler.Context{preMigrationCtx})
+
+		logrus.Infoln("Completed backup of apps destination cluster")
+
+		// Switch kubeconfig to source cluster for restore
+		err = setRemoteConfig("")
+		require.NoError(t, err, "Error resetting remote config")
+
+		srcBackupLocation, err := createBackupLocation(t, appKey+"-backup-location", testKey+"-mysql-migration", storkv1.BackupLocationType(location), secret)
+		require.NoError(t, err, "Error creating backuplocation %s", currBackupLocation.Name)
+
+		// Set sync to true on first cluster so that backup from second cluster is available
+		srcBackupLocation.Location.Sync = true
+		srcBackupLocation, err = storkops.Instance().UpdateBackupLocation(srcBackupLocation)
+		require.NoError(t, err, "Failed to set backup-location sync to true")
+
+		// Check periodically to see if the backup from this test is synced on second cluster
+		var allAppBackups *storkv1.ApplicationBackupList
+		listBackupsTask := func() (interface{}, bool, error) {
```

Comment:
This can be reused from the backup tests?

Reply:
Moved to a common method

Reply:
This file hasn't been updated with the call to those?

Reply:
Updated the file